lib/mcp/prompt.rb

```diff
@@ -8,13 +8,20 @@ class << self
       attr_reader :title_value
       attr_reader :description_value
       attr_reader :arguments_value
+      attr_reader :meta_value
 
       def template(args, server_context: nil)
         raise NotImplementedError, "Subclasses must implement template"
       end
 
       def to_h
-        { name: name_value, title: title_value, description: description_value, arguments: arguments_value.map(&:to_h) }.compact
+        {
+          name: name_value,
+          title: title_value,
+          description: description_value,
+          arguments: arguments_value&.map(&:to_h),
```

Comment:
It seems that the tests passes even without safe navigation. Is it necessary?

```suggestion
          arguments: arguments_value.map(&:to_h),
```